sql/core/src/main/scala/org/apache/spark/sql/execution/CollectMetricsExec.scala

```diff
@@ -98,13 +99,30 @@ object CollectMetricsExec extends AdaptiveSparkPlanHelper {
   /**
    * Recursively collect all collected metrics from a query tree.
    */
-  def collect(plan: SparkPlan): Map[String, Row] = {
-    val metrics = collectWithSubqueries(plan) {
+  def collect(physicalPlan: SparkPlan, analyzedOpt: Option[LogicalPlan]): Map[String, Row] = {
```

Comment:
when the `analyzedOpt` can be None? tests?

Reply:
Please take a look at the pattern for InMemoryTableScanExec: we may not have logical link for the physical plan.